packages/walletconnect/src/index.ts

```diff
@@ -125,15 +125,13 @@ export class WalletConnect extends Connector {
       await this.isomorphicInitialize()
       if (!this.provider?.connected) throw Error('No existing connection')
 
-      // for walletconnect, we always use sequential instead of parallel fetches because otherwise
-      // chainId defaults to 1 even if the connecting wallet isn't on mainnet
-      const accounts = await this.provider?.request<string[]>({ method: 'eth_accounts' })
+      // Wallets may resolve eth_chainId and hang on eth_accounts pending user interaction, which may include changing
+      // chains; they should be requested serially, with accounts first, so that the chainId can settle.
+      const accounts = await this.provider.request<string[]>({ method: 'eth_accounts' })
+      const chainId = await this.provider.request<string>({ method: 'eth_chainId' })
       if (!accounts.length) throw new Error('No accounts returned')
-      const chainId = await this.provider
```

Comment:
why this change? I actually commented in other place asking whether we shouldn't actually return early instead of letting `chainId` request to start when no accounts returned.